src/app/DatasetCounter.tsx

```diff
@@ -2,20 +2,35 @@
 //
 // SPDX-License-Identifier: Apache-2.0
 "use client";
+import { useAlert } from "@/providers/AlertProvider";
 import { datasetCount } from "@/services/discovery/index.public";
 import { useEffect, useState } from "react";
+import { AxiosError } from "axios";
 
 export function DatasetCounter() {
+  const { setAlert } = useAlert();
   const [count, setCount] = useState<number | null>(null);
 
   useEffect(() => {
     const fetchData = async () => {
-      const count = await datasetCount();
-      setCount(count);
+      try {
+        const count = await datasetCount();
+        setCount(count);
+      } catch (error) {
+        if (error instanceof AxiosError) {
+          setAlert({
+            type: "error",
+            message:
+              error.response?.data?.title ||
```

Comment:
suggestion: there is no need to add fallback for `title`, as we are implementing [RFC 7807](https://datatracker.ietf.org/doc/html/rfc7807).

`detail` is not mandatory, by this spec, but in our APIs we always have details.

`status` is mandatory and is always there as well.

Reply:
As agreed, it makes sense to have fallback title, if the exception is coming from another layer - e.g. Kong - and that layer does not implement this RFC.